lib/phoenix_live_view/lifecycle.ex

```diff
@@ -153,6 +153,14 @@ defmodule Phoenix.LiveView.Lifecycle do
   def mount(params, session, %Socket{private: %{@lifecycle => lifecycle}} = socket) do
     reduce_socket(lifecycle.mount, socket, fn %{id: {_mod, arg}} = hook, acc ->
       case hook.function.(arg, params, session, acc) do
+        {_, %Socket{redirected: to, private: %{render_with: render_with}}}
+        when is_function(render_with) and not is_nil(to) ->
+          raise_halt_with_redirect_and_render_with!(hook)
+
+        {:halt, %Socket{private: %{render_with: render_with}} = socket}
+        when is_function(render_with) ->
+          {:halt, put_private(socket, :halted, true)}
```

Comment:
Implementation wise, it would be best if we didn't have to look into both private/redirected. If we could unify in a single place, it would be better.